src/main/java/mate/academy/dao/impl/CinemaHallDaoImpl.java

```diff
@@ -40,19 +40,65 @@
         try (Session session = HibernateUtil.getSessionFactory().openSession()) {
             return Optional.ofNullable(session.get(CinemaHall.class, id));
         } catch (Exception e) {
-            throw new DataProcessingException("Can't get a cinema hall by id: " + id, e);
+            throw new DataProcessingException("Can't get a cinemaHall by id: " + id, e);
         }
     }
 
     @Override
     public List<CinemaHall> getAll() {
         try (Session session = HibernateUtil.getSessionFactory().openSession()) {
-            CriteriaQuery<CinemaHall> criteriaQuery = session.getCriteriaBuilder()
-                    .createQuery(CinemaHall.class);
-            criteriaQuery.from(CinemaHall.class);
-            return session.createQuery(criteriaQuery).getResultList();
+            return session.createQuery("from CinemaHall").list();
+        }
+    }
+
+    @Override
+    public Boolean update(CinemaHall cinemaHall) {
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+        Session session = null;
+        Transaction transaction = null;
+
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.update(cinemaHall);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("update " + cinemaHall + " failed", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+
+        return get(cinemaHall.getId()).get().equals(cinemaHall);
+    }
+
+    @Override
+    public Boolean delete(Long id) {
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+        Session session = null;
+        Transaction transaction = null;
+
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.delete(get(id)
+                    .orElseThrow(() -> new RuntimeException("no object with id " + id)));
+            transaction.commit();
         } catch (Exception e) {
-            throw new DataProcessingException("Can't get all cinema halls", e);
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("removing movie with id: "
+                    + id + " from database failed", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
         }
+        return get(id).isEmpty();
```

Comment:
The return statement in the `delete` method checks if the object is still present in the database. This can be inefficient. Consider returning `true` after a successful commit, assuming the transaction was successful.